gamemodes/terrortown/gamemode/server/sv_player_ext.lua

```diff
@@ -460,13 +462,17 @@ end
 -- @return boolean success?
 -- @realm server
 function plymeta:GiveEquipmentItem(id)
-	if self:HasEquipmentItem(id) then
-		return false
-	elseif id then
-		self:AddEquipmentItem(id)
+	if not id then return end
 
-		return true
+	local item = items.GetStored(id)
+
+	if item and item.limited and self:HasEquipmentItem(id) then
```

Comment:
what if item is nil?

Reply:
you're right. It is not checked here. However `self:AddEquipmentItem()` checks if the item is nil